util/sentry/sentry.cc

```diff
@@ -121,28 +126,52 @@ bool ParseDsn(const string& dsn, Dsn* res) {
   size_t kpos = dsn.find('@');
   if (kpos == string::npos)
     return false;
-  res->key = dsn.substr(0, kpos);
+  size_t protocol_sep_pos = dsn.find("://");
+  if (protocol_sep_pos == string::npos)
+    protocol_sep_pos = 0; // Nobody wrote http://, just assume it is http anyway.
+  else
+    protocol_sep_pos += 3;
+  std::string key = dsn.substr(protocol_sep_pos, kpos - protocol_sep_pos);
+  size_t colon_pos = key.find(':');
+  if (colon_pos == string::npos) {
+    res->public_key = key;
+    res->secret_key = "";
+  } else {
+    res->public_key = key.substr(0, colon_pos);
+    res->secret_key = key.substr(colon_pos + 1);
+  }
   ++kpos;
   size_t pos = dsn.find('/', kpos);
-  if (kpos == string::npos)
+  if (pos == string::npos)
     return false;
   res->hostname = dsn.substr(kpos, pos - kpos);
   res->url = dsn.substr(pos);
 
-  VLOG(1) << "Dsn is: " << res->key << "|" << res->hostname << "|" << res->url;
+  VLOG(1) << "Dsn is: " << res->public_key
+          << "|" << res->secret_key
+          << "|" << res->hostname
+          << "|" << res->url;
   return true;
 }
 
 }  // namespace
 
 void EnableSentry(IoContext* context) {
-  if (FLAGS_sentry_dsn.empty()) {
-    LOG(INFO) << "--sentry_dsn is not defined, sentry is disabled";
+  static std::atomic<bool> ran_once(false);
+  CHECK(!ran_once.exchange(true)) << "EnableSentry called twice";
+
+  std::string sentry_dsn = FLAGS_sentry_dsn;
+  if (sentry_dsn.empty()) {
```

Comment:
in most cases we do not use sentry but now we will call this function. I suggest that instead of printing what we did not find, print what you succeeded to find...

Reply:
Or move it to vlog(1)